extensions/amp-analytics/0.1/test/test-requests.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {installVariableService, ExpansionOptions} from '../variables';
+import {expandConfigRequest, RequestHandler} from '../requests';
+import {macroTask} from '../../../../testing/yield';
+import * as lolex from 'lolex';
+
+
+describes.realWin('Requests', {amp: 1}, env => {
+  let win;
+  let ampdoc;
+  let clock;
+
+  beforeEach(() => {
+    installVariableService(env.win);
+    win = env.win;
+    ampdoc = env.ampdoc;
+    clock = lolex.install(win);
+  });
+
+  describe('RequestHandler', () => {
+    describe('batch Delay', () => {
+      it('maxDelay should  be a number', () => {
+        const r1 = {'baseUrl': 'r1', 'maxDelay': 1};
+        const r2 = {'baseUrl': 'r2', 'maxDelay': '2'};
+        const r3 = {'baseUrl': 'r3', 'maxDelay': '2.5'};
+        const r4 = {'baseUrl': 'r4', 'maxDelay': 'invalid'};
+        const r5 = {'baseUrl': 'r5'};
+        const handler1 = new RequestHandler(ampdoc, r1, () => {}, false);
+        const handler2 = new RequestHandler(ampdoc, r2, () => {}, false);
+        const handler3 = new RequestHandler(ampdoc, r3, () => {}, false);
+        const handler4 = new RequestHandler(ampdoc, r4, () => {}, false);
+        const handler5 = new RequestHandler(ampdoc, r5, () => {}, false);
+        expect(handler1.maxDelay_).to.equal(1);
+        expect(handler2.maxDelay_).to.equal(2);
+        expect(handler3.maxDelay_).to.equal(2.5);
+        expect(handler4.maxDelay_).to.equal(0);
+        expect(handler5.maxDelay_).to.equal(0);
```

Comment:
Error case? Can we match log statements?

Reply:
I don't think we need to. The log comes from the `#Number()` function, and is not something we want to test here.